cdisc_rules_engine/utilities/data_processor.py

```diff
@@ -262,121 +200,184 @@ def valid_meddra_term_references(self, dataframe, target, domain, **kwargs):
+                    f"{params.domain}{MedDRAVariables.HLTCD.value}",
+                    f"{params.domain}{MedDRAVariables.HLT.value}",
                 ),
             ),
-            f"{domain}{MedDRAVariables.DECOD.value}": (
+            f"{params.domain}{MedDRAVariables.DECOD.value}": (
                 TermTypes.PT.value,
                 (
-                    f"{domain}{MedDRAVariables.PTCD.value}",
-                    f"{domain}{MedDRAVariables.DECOD.value}",
+                    f"{params.domain}{MedDRAVariables.PTCD.value}",
+                    f"{params.domain}{MedDRAVariables.DECOD.value}",
                 ),
             ),
-            f"{domain}{MedDRAVariables.PTCD.value}": (
+            f"{params.domain}{MedDRAVariables.PTCD.value}": (
                 TermTypes.PT.value,
                 (
-                    f"{domain}{MedDRAVariables.PTCD.value}",
-                    f"{domain}{MedDRAVariables.DECOD.value}",
+                    f"{params.domain}{MedDRAVariables.PTCD.value}",
+                    f"{params.domain}{MedDRAVariables.DECOD.value}",
                 ),
             ),
-            f"{domain}{MedDRAVariables.LLT.value}": (
+            f"{params.domain}{MedDRAVariables.LLT.value}": (
                 TermTypes.LLT.value,
                 (
-                    f"{domain}{MedDRAVariables.LLTCD.value}",
-                    f"{domain}{MedDRAVariables.LLT.value}",
+                    f"{params.domain}{MedDRAVariables.LLTCD.value}",
+                    f"{params.domain}{MedDRAVariables.LLT.value}",
                 ),
             ),
-            f"{domain}{MedDRAVariables.LLTCD.value}": (
+            f"{params.domain}{MedDRAVariables.LLTCD.value}": (
                 TermTypes.LLT.value,
                 (
-                    f"{domain}{MedDRAVariables.LLTCD.value}",
-                    f"{domain}{MedDRAVariables.LLT.value}",
+                    f"{params.domain}{MedDRAVariables.LLTCD.value}",
+                    f"{params.domain}{MedDRAVariables.LLT.value}",
                 ),
             ),
         }
-        term_type, columns = variable_pair_map.get(target)
-        cache_key = f"meddra_valid_code_term_pairs_{dictionaries_path}"
+        term_type, columns = variable_pair_map.get(params.target)
+        cache_key = f"meddra_valid_code_term_pairs_{params.meddra_path}"
         valid_code_term_pairs = self.cache.get(cache_key)
         if not valid_code_term_pairs:
-            terms: dict = self.cache.get(dictionaries_path)
+            terms: dict = self.cache.get(params.meddra_path)
             valid_code_term_pairs = MedDRATerm.get_code_term_pairs(terms)
             self.cache.add(cache_key, valid_code_term_pairs)
         column = str(uuid4()) + "_pairs"
-        dataframe[column] = list(zip(dataframe[columns[0]], dataframe[columns[1]]))
-        result = dataframe[column].isin(valid_code_term_pairs[term_type])
+        params.dataframe[column] = list(
```

Comment:
[we can use list comprehension and save couple msec](https://stackoverflow.com/questions/5790860/and-vs-list-and-dict-which-is-better), but it more of coding style for me

Reply:
Not sure:

```
timeit.timeit("list(zip([1, 2, 3], [4, 5, 6]))")
Out[9]: 0.4584283690000035
timeit.timeit("[x for x in zip([1, 2, 3], [4, 5, 6])]")
Out[10]: 0.5141287360000035
```